### ISSUE TITLE
Wemo Light Switch Long Press

### ISSUE THREAD

**Issue by JideOsan** [iancmcc/ouimeaux#60]
Hey,
I just got the Wemo light switch. A few months ago Belkin added a long press feature. I've been digging through the SOAP api and I cant find the XML to control/detect when it is pressed. Have you had any luck? It would be great if you can add support for this to project.

**Comment by iancmcc** [comment]
I haven't found this yet, but I'll investigate further.

**Comment by nexus-bytes** [comment]
I too would love to have support for the light switch long press. However, according to comments on the Wemo discussion board, it seems that the long press does not have an associated UPnP event (just an IFTTT event.

(See post by MikeP: http://community.wemothat.com/t5/WEMO-Hardware/WEMO-Software-Product-Manager-An-Introduction/m-p/7391/highlight/true#M452 and resposne by Tom_Hudson - Wemo Software Product Manager: http://community.wemothat.com/t5/WEMO-Hardware/WEMO-Software-Product-Manager-An-Introduction/m-p/7395/highlight/true#M454).

There is hope that Belkin will add the UPnP event at some point in the future...

**Comment by dlamkins** [comment]
I could be wrong, but based on my understanding of the eventservice.xml file hosted by the WeMo lightswitches, it looks like they now support the long press event via UPnP.  My switches are running firmware version "WeMo_WW_2.00.10937.PVT-OWRT-LS" which I got on Jan 25, 2017.  Here is my eventservice.xml contents:

[http://pastebin.com/zJgesnAc](http://pastebin.com/zJgesnAc)

I've highlighted the two sections regarding long presses.  The first looks to be for simulating a long press on the WeMo and the second appears to be an event that can be subscribed to.  Here are those sections pulled out:

**Action (In Action List)**

```
<action>
<name>SimulateLongPress</name>
<argumentList>
<argument>
<name>LongPress</name>
<direction>out</direction>
<relatedStateVariable>LongPress</relatedStateVariable>
</argument>
</argumentList>
</action>
```

**Event (In Service State Table)**

```
<stateVariable sendEvents="yes">
<name>LongPress</name>
<dataType>string</dataType>
<defaultValue>0</defaultValue>
</stateVariable>
```

Is it possible that this feature could get added?
